### PR TITLE
replica/database: drop_column_family(): properly cleanup stale querier cache entries

### PR DESCRIPTION
**Pull request by denesb**
Said method has to evict all querier cache entries, belonging to the to-be-dropped table. This is already the case, but there was a window where new entries could sneak in, causing a stale reference to the table to be de-referenced later when they are evicted due to TTL. This window is now closed, the entries are evicted after the method has waited for all ongoing operations on said table to stop.

Fixes: #10450

### CODE REVIEW

File: replica/database.cc [hunk 1]
```diff
@@ -929,11 +930,10 @@ future<> database::drop_column_family(const sstring& ks_name, const sstring& cf_
...
     cf->clear_views();
-    co_return co_await cf->await_pending_ops().then([this, &ks, cf, tsf = std::move(tsf), snapshot] {
-        return truncate(ks, *cf, std::move(tsf), snapshot).finally([this, cf] {
-            return cf->stop();
-        });
-    }).finally([cf] {});
+    co_await cf->await_pending_ops();
+    auto f = co_await coroutine::as_future(truncate(ks, *cf, std::move(tsf), snapshot));
+    co_await cf->stop();
+    f.get(); // re-throw exception from truncate() if any
```
Comment (bhalevy) [inline replica/database.cc:936]: it's a shame we can't simply co_return the future... but this will work too.

**Review by bhalevy — APPROVED** [review]
LGTM

File: replica/database.cc (hunk 1 above)
Comment (bhalevy) [reply replica/database.cc:936]: I wonder if `co_return co_await std::move(f);` will save the exception rethrowing.

Comment (denesb) [reply replica/database.cc:936]: maybe, but how is that prettier :))

File: replica/database.cc [hunk 2]
```diff
@@ -929,11 +930,10 @@ future<> database::drop_column_family(const sstring& ks_name, const sstring& cf_
...
     co_await remove(*cf);
     cf->clear_views();
-    co_return co_await cf->await_pending_ops().then([this, &ks, cf, tsf = std::move(tsf), snapshot] {
-        return truncate(ks, *cf, std::move(tsf), snapshot).finally([this, cf] {
-            return cf->stop();
-        });
-    }).finally([cf] {});
+    co_await cf->await_pending_ops();
+    auto f = co_await coroutine::as_future(truncate(ks, *cf, std::move(tsf), snapshot));
+    co_await cf->stop();
```
Comment (avikivity) [inline replica/database.cc:935]: This will back backports harder, but I don't mind.

**Comment by avikivity** [comment]
Looks good.

File: replica/database.cc (hunk 2 above)
Comment (denesb) [reply replica/database.cc:935]: We only have to backport the last patch, the rest is cosmetic.

File: replica/database.cc (hunk 1 above)
Comment (bhalevy) [reply replica/database.cc:936]: it does throw, so no point in doing that.

**Pull request merged**